### PR TITLE
Add initial integration tests

### PR DESCRIPTION
**Pull request by tyll**

### CODE REVIEW

File: tests/tests_basic.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+# empty playbook to gather facts for import_playbook when clause
+- hosts: all
+
+# workaroud for: https://github.com/ansible/ansible/issues/27973
```
Comment (phoracek) [inline tests/tests_basic.yml]: Could you put here tl;dr version of the problem?

File: tests/playbooks/centos7_deps.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+---
+- name: Install CentOS 7 deps for nmstate tests
+  hosts: all
+  vars:
+    - rpmdependencies:
+      - git
+      - python2-pip
+      - NetworkManager
```
Comment (phoracek) [inline tests/playbooks/centos7_deps.yml]: NetworkManager is not installed by default?

File: tests/README-integration-tests.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+The integration tests are written as Ansible playbooks targetted for all hosts.
+They follow the [Fedora's Standard Test
+Interface](https://fedoraproject.org/wiki/CI/Standard_Test_Interface) and can
+be run by the [Linux Sytem Roles Test
+Harness](https://github.com/linux-system-roles/test-harness). You can also run
+them directly, for example to run them against the [CentOS Cloud
+Images](https://cloud.centos.org/centos/7/images/) download one of them,
+install `standard-test-roles` from Fedora and execute `ansible-playbook`:
+
```
Comment (phoracek) [inline tests/README-integration-tests.md]: Could you add more meat here? Just a short description of what are added files and subdirectories good for.

File: tests/ci_tests_basic.yml [hunk 4]
```diff
@@ -0,0 +1,83 @@
+---
+- name: empty play to gather facts
+  hosts: all
+- import_playbook: playbooks/rhel7_deps.yml
```
Comment (phoracek) [inline tests/ci_tests_basic.yml]: please add empty lines between plays

File: tests/ci_tests_basic.yml [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    - ansible_distribution_major_version == '7'
+- name: Run tox tests and nmstatectl show
+  hosts: all
+  vars:
+    - nmstatedir: /tmp/nmstate-git-checkout
+  tasks:
+    # Does not work when specifying multiple hosts:
+    # https://github.com/ansible/ansible/issues/27973
+    # do not run on CentOS/RHEL 6
+    #- meta: end_play
```
Comment (phoracek) [inline tests/ci_tests_basic.yml]: Could you either add a TODO or remove the code? I personally don't like commented-out code.

File: tests/ci_tests_basic.yml [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+    - synchronize:
+        src: ../../
+        dest: "{{ nmstatedir }}"
+        recursive: yes
+        delete: yes
+        # tox reqires it to be a .git directory, so cannot exclude .git
+        rsync_opts:
+          - "--exclude=.pyc"
+          - "--exclude=__pycache__"
+      when: false
```
Comment (phoracek) [inline tests/ci_tests_basic.yml]: what does this do?

File: tests/ci_tests_basic.yml [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+
+    - pip:
+        name: tox
+        extra_args: --user
+
+    - command: ~/.local/bin/tox
+      args:
+        chdir: "{{ nmstatedir }}"
+
+    # this needs https://github.com/ansible/ansible/pull/40793 to be merged
```
Comment (phoracek) [inline tests/ci_tests_basic.yml]: Please add TODO: to the beggining. Is this blocking merging of this patch?

File: tests/README-integration-tests.md [hunk 8]
```diff
@@ -0,0 +1,15 @@
+# Integration Tests #
```
[the remark this answers is not in the document]
Comment (tyll) [reply tests/README-integration-tests.md]: The Standard Test Interface requires that the tests are directly in the `tests/` directory. I would like to execute thet test from ctl/ and lib/, too, but I did not want to start to figure out how to get them running on RHEL 7 (e.g. which repos to enable etc).

File: tests/tests_basic.yml (hunk 1 above)
Comment (tyll) [reply tests/tests_basic.yml]: done

File: tests/playbooks/centos7_deps.yml (hunk 2 above)
Comment (tyll) [reply tests/playbooks/centos7_deps.yml]: It is not in cloud images.

File: tests/README-integration-tests.md (hunk 3 above)
Comment (tyll) [reply tests/README-integration-tests.md]: done

File: tests/ci_tests_basic.yml (hunk 4 above)
Comment (tyll) [reply tests/ci_tests_basic.yml]: done

File: tests/ci_tests_basic.yml (hunk 5 above)
Comment (tyll) [reply tests/ci_tests_basic.yml]: done

File: tests/ci_tests_basic.yml (hunk 6 above)
Comment (tyll) [reply tests/ci_tests_basic.yml]: synchronize is currently broken in Fedora updates-testing therefore the tar workaround is used (I updated the comments)

File: tests/ci_tests_basic.yml (hunk 7 above)
Comment (tyll) [reply tests/ci_tests_basic.yml]: No, there is another workaround for the broken pip module in the next task, I added comments as well.

**Comment by tyll** [comment]
The tests work now after we merged this I will enable it on the automatic CI and we can start adding more complex tests.

**Comment by EdDev** [comment]
I would like to see pytest used to run the integration tests and Ansible be left with just setting up the tested host.

**Pull request merged**